fms/models/hf/utils.py

```diff
@@ -130,3 +135,88 @@ def to_hf_api(model: nn.Module, **override_config_kwargs) -> "HFModelArchitectur
 
     hf_adapted_cls = _fms_to_hf_adapt_map[model_type]
     return hf_adapted_cls.from_fms_model(model, **override_config_kwargs)
+
+
+def as_fms_model(
+    model_id_or_path: Union[str, os.PathLike], device: Union[str, torch.device] = "cpu"
+) -> nn.Module:
+    """
+    get an FMS model from a huggingface checkpoint
+
+    Parameters
+    ----------
+    model_id_or_path: Union[str, os.PathLike]
+        The huggingface hub model id or a local path. If the local path exists, the model will be loaded directly from
+        the local path, otherwise the huggingface cache will be checked. If the huggingface cache does not contain the
+        model, then the weights will be downloaded and stored into the huggingface cache
+    device: Union[str, torch.device]
+        the device to load the model weights to
+
+    Returns
+    -------
+    nn.Module
+        an fms equivalent implementation of an HF model
+    """
+    if not os.path.exists(model_id_or_path):
+        model_id_or_path = snapshot_download(repo_id=model_id_or_path)
+
+    config = AutoConfig.from_pretrained(model_id_or_path)
+
+    architecture = config.architectures[0]
+    params = {
+        "model_path": model_id_or_path,
+        "source": "hf",
+        "device_type": device.type if isinstance(device, torch.device) else device,
+    }
```

Comment:
nit: we might want to consider these explicitly to get_model as regular params. I think we can mix regular kwargs and **kwargs.

Also I think the results of `snapshot_download` will alwyas be a path and never a model_id? maybe we should call it as such, not re-use `model_id_or_path`

Reply:
Fixed in latest commit